wheatley/rhythm.py

```diff
@@ -155,6 +164,11 @@ def initialise_line(self, stage, user_controls_treble, start_time,
 
         self._inner_rhythm.initialise_line(stage, user_controls_treble, start_time - self.delay,
                                            number_of_user_controlled_bells)
+        self._expected_bells[HANDSTROKE].clear()
+        self._expected_bells[BACKSTROKE].clear()
+        self._early_bells[HANDSTROKE].clear()
+        self._early_bells[BACKSTROKE].clear()
+        self._current_stroke = HANDSTROKE
```

Comment:
I think this is the part that has been causing the issues with Wheatley just not ringing.

Shouldn't `self.delay` also be reset to `0` here?  If it should (and currently isn't), then it would explain the behaviour I've been seeing, since the cumulative delay of the last touch would be applied to the first row of the next touch, causing Wheatley to appear to hang.

Reply:
> Shouldn't `self.delay` also be reset to 0 here?

As it's just a constant time shift applied when passing times to the `inner_rhythm` I don't think it makes any difference, but I'll step through it properly.  It could be set to 0 if it assumes the `inner_rhythm` also completely resets here.

My hypothesis for hanging behaviour is in a fireup and then all bells being reset to handstroke Wheatley is in a loop waiting for a bell to ring (say the 7th). Clearing `_expected_bells` should get it out of the current wait and ready again on `LOOK TO`

Reply:
Actually I think the case where it hangs in a loop would result in a large delay being added at the wrong point here, though that's an awful race condition to try and sort out. Hmmm